api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -517,7 +517,10 @@ func (s *SolutionManager) Reconcile(ctx context.Context, deployment model.Deploy
 				summary.AllAssignedDeployed = false
 				targetResultStatus := fmt.Sprintf("%s Failed", deploymentType)
 				summary.UpdateTargetResult(step.Target, model.TargetResultSpec{Status: targetResultStatus, ComponentResults: componentResults}) // TODO: this keeps only the last error on the target
-				time.Sleep(5 * time.Second)                                                                                                     //TODO: make this configurable?
+				if !v1alpha2.IsRetriableErr(stepError) {
```

Comment:
i think we need to take non-coa error into consideration

Reply:
created `IsRetriableErr` for `APIError`. Any other error types which has implemented function `IsRetriableErr()` can use this helper function to determine if it is retriable.